.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -1 +1,32 @@
-<!-- Please provide a clear and meaningful description to the CHANGELOG.md file if this PR contributes some significant changes -->
+<!--
+Thank you for your Pull Request. Please provide a description above and review
+the requirements below.
+
+Bug fixes and new features should include tests and possibly benchmarks.
+
+Contributors guide: https://github.com/theia-ide/theia/blob/master/CONTRIBUTING.md
+-->
+
+#### What it does
+<!-- Include relevant issues and describe how they are addressed -->
+
+#### How to test
+<!-- Describe how a reviewer can verify changes within Theia repo -->
+
+#### Review Checklist
+<!-- For completed items, change [ ] to [x].  -->
+<!-- For inapplicable items, cross out with `[ ] foo` to `[x] ~~foo~~`.  -->
+
+- [ ] commits are signed-off: https://github.com/theia-ide/theia/blob/master/CONTRIBUTING.md#sign-your-work
+- [ ] commit history is cleaned up of merge and fix up commits
+  - use `git pull -r` to pick up changes from the master
+- [ ] tests/benchmarks are defined for functionallity/perf optimizations and verified by a reviewer, not an author
```

Comment:
Does this mean we have to have unit tests? If yes, what coverage? What about things that are easier to test manually (typos, adding a UI element like menu item)?

Reply:
Can we leave this out?

Reply:
I don't want to make the process overly strict and did not mean require unit tests for each PR.

I meant that each PR should at least explain how to reproduce a bug, test new functionality, verify perf improvements. If someone wants to add unit or UI tests it is always welcomed, but not mandatory.

I will remove it and add `A contributor should explain how to reproduce a bug, test new functionality or verify perf improvements.` as a comment under `How to test` section.